### PR TITLE
tp-qemu: remove vdi/qed format from disk format test

### PR DESCRIPTION
**Pull request by hellohellenmao**
ID: 1664937

### CODE REVIEW

File: qemu/tests/cfg/qemu_disk_img.cfg [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # For windows guest, you can download md5sum.exe from http://www.brothersoft.com/md5sum-81781.html,
 # then add parents dir to $PATH or config md5sum in share/cfg/guest-os/Windows.cfg;
 - qemu_disk_img:
-    no qed raw vmdk
```
Comment (hereischen) [inline qemu/tests/cfg/qemu_disk_img.cfg]: should keep `no qed`

File: qemu/tests/cfg/qemu_disk_img.cfg [hunk 2]
```diff
@@ -2,7 +2,7 @@
...
+    no raw vmdk
     virt_test_type = qemu
```
Comment (pingl2016) [inline qemu/tests/cfg/qemu_disk_img.cfg]: It looks like that you want to support qed format for this case.

File: qemu/tests/cfg/qemu_img.cfg [hunk 3]
```diff
@@ -12,7 +12,7 @@
             remove_image_dd = yes
             create_image_cmd = "dd if=/dev/zero of=%s bs=1G count=1"
             # Test the conversion from 'dd_image_name' to specified format
-            supported_image_formats = qcow2 raw qed
+            supported_image_formats = qcow2 raw
             Host_RHEL.m5:
```
Comment (pingl2016) [inline qemu/tests/cfg/qemu_img.cfg]: As supported_image_formats is same, no need to separate it for RHEL-5 host. Drop the follow two lines.

File: qemu/tests/cfg/qemu_img.cfg [hunk 4]
```diff
@@ -77,18 +77,14 @@
...
-                    dest_image_format = qed
 
         - snapshot:
             subcommand = snapshot
             only qcow2
         - info:
             subcommand = info
         - rebase:
-            no raw vmdk qed
+            no raw vmdk
```
Comment (pingl2016) [inline qemu/tests/cfg/qemu_img.cfg]: It looks like that you want to support qed format for this case.

File: qemu/tests/cfg/qemu_img.cfg [hunk 5]
```diff
@@ -106,7 +102,7 @@
                             check_output = stderr
         # Commit is the only subtest that does need an installed guest
         - commit:  install setup image_copy unattended_install.cdrom
-            no vmdk raw qed
+            no vmdk raw
```
Comment (pingl2016) [inline qemu/tests/cfg/qemu_img.cfg]: It looks like that you want to support qed format for this case.

File: qemu/tests/cfg/qemu_iotests.cfg [hunk 6]
```diff
@@ -19,13 +19,6 @@
         - qcow2_format:
             only qcow2 qcow2v3
             qemu_io_image_format = qcow2
-        - qed_format:
```
Comment (pingl2016) [inline qemu/tests/cfg/qemu_iotests.cfg]: This case is for upstream qemu. Please make sure the formats you dropped are deprecated or they won't be used by upstream customers.

File: qemu/tests/cfg/qemu_disk_img.cfg (hunk 2 above)
Comment (hellohellenmao) [reply qemu/tests/cfg/qemu_disk_img.cfg]: 'qed' is compiled out of qemu package, so I think there should be no 'qed' any more. Like 'abc', 'hisd', ... which formats are not supported.

File: qemu/tests/cfg/qemu_iotests.cfg (hunk 6 above)
Comment (hellohellenmao) [reply qemu/tests/cfg/qemu_iotests.cfg]: Thanks for your info. I think upstream still use it for there is still codes update about qed/vdi format from the upstream code. And I will roll back the changes.

File: qemu/tests/cfg/qemu_img.cfg (hunk 3 above)
Comment (hellohellenmao) [reply qemu/tests/cfg/qemu_img.cfg]: Yes, thanks.

File: qemu/tests/cfg/qemu_iotests.cfg [hunk 7]
```diff
@@ -19,7 +19,7 @@
...
+	- qed_format:
```
Comment (pingl2016) [inline qemu/tests/cfg/qemu_iotests.cfg]: Please replace tab with blanks.

File: qemu/tests/cfg/qemu_disk_img.cfg (hunk 2 above)
Comment (pingl2016) [reply qemu/tests/cfg/qemu_disk_img.cfg]: How did you know that? Could you share the bug or documentation?

Comment (hellohellenmao) [reply qemu/tests/cfg/qemu_disk_img.cfg]: The info is from Kevin, you could check the description from the related bug.

Comment (pingl2016) [reply qemu/tests/cfg/qemu_disk_img.cfg]: The description is not suitable.
The reason why QEMU upstream develop QED is to create a better perfoming image format by removing some features compared to qcow2(such as refcount table). It is indeed a official image format. But for downstream qemu, i am not sure. And even for upstream qemu, the development of QED has been abandoned and the improvements have been merged back into qcow2.

Comment (hellohellenmao) [reply qemu/tests/cfg/qemu_disk_img.cfg]: An other fact is that we could not create qed/vdi image file in latest qemu. It prompts the format is not supported.

[root@lenovo-sr630-02 ~]# qemu-img create -f qed test.img 1G
qemu-img: test.img: Unknown file format 'qed'
[root@lenovo-sr630-02 ~]# qemu-img create -f vdi test.img 1G
qemu-img: test.img: Unknown file format 'vdi'

Comment (pingl2016) [reply qemu/tests/cfg/qemu_disk_img.cfg]: I knew that. But your commit message is not suitable to drop a feature in tp-qemu. As the repo is for upstream testing. If all your changes are only for downstream qemu, the correct way is remove the case in internal test loops.

**Comment by hellohellenmao** [comment]
Updated, thanks @pingl2016

**Pull request merged**